src/Microsoft.OpenApi/Services/OpenApiFilterService.cs

```diff
@@ -124,11 +171,137 @@ public static OpenApiDocument CreateFilteredDocument(OpenApiDocument source, Fun
             return subset;
         }
 
-        private static IList<SearchResult> FindOperations(OpenApiDocument graphOpenApi, Func<OpenApiOperation, bool> predicate)
+        /// <summary>
+        /// Creates an <see cref="OpenApiUrlTreeNode"/> from a collection of <see cref="OpenApiDocument"/>.
+        /// </summary>
+        /// <param name="sources">Dictionary of labels and their corresponding <see cref="OpenApiDocument"/> objects.</param>
+        /// <returns>The created <see cref="OpenApiUrlTreeNode"/>.</returns>
+        public static OpenApiUrlTreeNode CreateOpenApiUrlTreeNode(Dictionary<string, OpenApiDocument> sources)
+        {
+            var rootNode = OpenApiUrlTreeNode.Create();
+            foreach (var source in sources)
+            {
+                rootNode.Attach(source.Value, source.Key);
+            }
+            return rootNode;
+        }
+
+        /// <summary>
+        /// Takes in a file stream, parses the stream into a JsonDocument and gets a list of paths and Http methods
+        /// </summary>
+        /// <param name="stream"> A file stream.</param>
+        /// <returns> A dictionary of request urls and http methods from a collection.</returns>
+        public static Dictionary<string, List<string>> ParseJsonCollectionFile(Stream stream)
+        {
+            var requestUrls = new Dictionary<string, List<string>>();
+
+            // Convert file to JsonDocument
+            using JsonDocument document = JsonDocument.Parse(stream);
```

Comment:
Consider using implicit type `var` here and subsequent lines below, to maintain consistency.